src/app/code/community/Omise/Gateway/Model/Payment/Creditcard.php

```diff
@@ -1,32 +1,19 @@
 <?php
-class Omise_Gateway_Model_Payment_Creditcard extends Omise_Gateway_Model_Payment
+class Omise_Gateway_Model_Payment_Creditcard extends Omise_Gateway_Model_Payment_Base_Payment
 {
-    /**
-     * @var string
-     */
-    protected $_code = 'omise_gateway';
```

Comment:
I think it is better to leave these and remove the getters and setters as mentioned in other comments

Reply:
We then also remove the need for the constructor method